client/components/common/tce-core/utils.js

```diff
@@ -1,13 +1,21 @@
 import toCase from 'to-case';
 
 export function getComponentName(type) {
-  return `tce-${toCase.slug(type)}`;
+  return `tce-${toCase.slug(getElementType(type))}`;
 }
 
 export function processAssessmentType(type) {
   return `assessment-${toCase.slug(type)}`;
 }
 
+export function isQuestion(type) {
+  return ['QUESTION', 'REFLECTION', 'ASSESSMENT'].includes(type);
+}
+
+export function getElementType(type) {
```

Comment:
Rename to processElementType / resolveElementType and document the purpose (Might have common wrapper for multiple element types). We also might consider renaming root element to something other since Assessment is a sub-class.